modules/c++/nitf/unittests/test_tre_mods++.cpp

```diff
@@ -117,6 +117,32 @@ TEST_CASE(populateWhileIterating)
     }
     TEST_ASSERT_EQ(numFields, 29);
 }
+
+TEST_CASE(overflowingNumericFields)
+{
+    nitf::TRE tre("CSCRNA");
+
+    // This field has a length of 9, so check that it's properly
+    // truncated
+    tre.setField("ULCNR_LAT", 1.0 / 9);
+    TEST_ASSERT_EQ(tre.getField("ULCNR_LAT").toString(), "0.1111111");
+
+    tre.setField("ULCNR_LAT", 123456789);
+    TEST_ASSERT_EQ(tre.getField("ULCNR_LAT").toString(), "123456789");
+
+    tre.setField("ULCNR_LAT", 12345678.);
+    TEST_ASSERT_EQ(tre.getField("ULCNR_LAT").toString(), "012345678");
```

Comment:
Why does a 0 get appended to the beginning?

Reply:
Because the field must be exactly 9 characters. Space is not a valid character for BCS-N (i.e. numeric) fields, so they get 0-padded from the left.